pkg/translate/translate_value.go

```diff
@@ -267,26 +263,58 @@ func translateHPASpec(inPath string, outPath string, value interface{}, valueTre
 	if !asEnabled {
 		return nil
 	}
-	newP := util.PathFromString(inPath)[0]
-	minPath := newP + ".autoscaleMin"
-	asMin, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(minPath))
-	if found && err == nil {
-		if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath+".minReplicas"), asMin); err != nil {
+
+	newP := util.PathFromString(inPath)
+	// last path element is k8s setting name
+	newPS := newP[:len(newP)-1].String()
```

Comment:
could write above two lines as
newPS := util.PathFromString(inPath)[len(newP)-1]

Reply:
the newP is defined from util.PathFromString(inPath)
so for one line, we have to write util.PathFromString(inPath)[:len(util.PathFromString(inPath)-1].String(), which is too lengthy